src/stdune/path.ml

```diff
@@ -1,14 +1,41 @@
+let is_dir_sep =
+  match Sys.os_type with
+  | "Win32" | "Cygwin" ->
+    fun c -> c = '/' || c = '\\' || c = ':'
+  | _ -> fun c -> c = '/'
+
 let explode_path =
-  let rec loop path acc =
-    let dir  = Filename.dirname  path in
-    let base = Filename.basename path in
-    let acc = base :: acc in
-    if dir = Filename.current_dir_name then
+  let rec start acc path i =
+    if i < 0 then
       acc
+    else if is_dir_sep (String.unsafe_get path i) then
+      start acc path (i - 1)
     else
-      loop dir acc
+      component acc path i i
```

Comment:
Why not `component acc path i (i - 1)`? It feels like we are scanning the same character twice otherwise